New cv for sampling two molecule distance; Parmchk2 for missing bonded potential

franklhy opened the pull request:
1. Add new cv and python script for sampling two molecules' distance
2. Use parmchk2 to handle missing bonded potentials.

franklhy committed:
New CV for distance between two molecules.

franklhy committed:
Add docstring to new CV

franklhy committed:
1. Add python script for sampling the distance between two motif molecules.
2. Using parmchk2 to find missing bonded potentials

franklhy committed:
Remove force fields for ny gaff2 genereted by ourselves, because parmchk2 already handle that.

p-zubieta commented on silc/md/collective_variables.py:
```diff
@@ -127,3 +128,43 @@ def com_angle(r1, r2, r3):
+        return lambda r1, r2: distance_pbc(r1, r2, self.box)
+
+
+def periodic(distance, box):
+    return np.mod(distance + box * 0.5, box) - 0.5 * box
+
+
+def distance_pbc(r1, r2, box):
+    r1 = barycenter(r1)
+    r2 = barycenter(r2)
+    dr = r1 - r2
+    return linalg.norm(periodic(dr, box))
```
```suggestion
def wrap(distance, box):
    return np.mod(distance + box * 0.5, box) - 0.5 * box


def distance_pbc(r1, r2, box):
    r1 = barycenter(r1)
    r2 = barycenter(r2)
    dr = r1 - r2
    return linalg.norm(wrap(dr, box))

```
Maybe wrap is a better name here.

franklhy committed:
Update silc/md/collective_variables.py

Co-authored-by: Pablo Zubieta <8410335+pabloferz@users.noreply.github.com>

franklhy committed:
(1) Add CV: alignment between a rod (motif) and a plate (gamma-CD).
(2) Add instruction on installing silc and pysages on midway2 to README.
(3) Add the python sscript to calculate the free energy of stacking between two motif molecules.
(4) Add python scripts to calculate the free energy of binding between one motif and gamma-CD.

franklhy committed:
rename function

franklhy committed:
Merge branch 'new_cv_2mol' of https://github.com/franklhy/silc into new_cv_2mol
